### PR TITLE
[FIX] Additional metrics during training

### LINKED ISSUE #101: Additional metrics in the trainer
During training, there is support to calculate the epoch score during each iteration. 

The optimization metric provided by the user is not added there automatically. We should prefer showing the optimization metric rather than always accuracy.

### PR DESCRIPTION
**Pull request by ravinkohli**
This PR fixes #101

### CODE REVIEW

File: autoPyTorch/evaluation/abstract_evaluator.py [hunk 1]
```diff
@@ -541,13 +519,65 @@ def __init__(self, backend: Backend,
...
+            'X_train': self.X_train,
+            'y_train': self.y_train,
+            'X_test': self.X_test,
+            'y_test': self.y_test,
+            'backend': self.backend,
+            'logger_port': logger_port,
+            'optimize_metric': self.metric.name
+        })
+
+        self.fit_dictionary = {**pipeline_config, **self.fit_dictionary}
```
Comment (nabenabe0928) [inline autoPyTorch/evaluation/abstract_evaluator.py:570]:
```suggestion
        self.fit_dictionary.update(pipeline_config)
```

File: autoPyTorch/evaluation/abstract_evaluator.py [hunk 2]
```diff
@@ -541,13 +519,65 @@ def __init__(self, backend: Backend,
...
+        # If the budget is epochs, we want to limit that in the fit dictionary
+        if self.budget_type == 'epochs':
+            self.fit_dictionary['epochs'] = self.budget
+            self.fit_dictionary.pop('runtime', None)
+        elif self.budget_type == 'runtime':
+            self.fit_dictionary['runtime'] = self.budget
+            self.fit_dictionary.pop('epochs', None)
+        else:
+            raise ValueError(f"Unsupported budget type {self.budget_type} provided")
```
Comment (nabenabe0928) [inline autoPyTorch/evaluation/abstract_evaluator.py]:
```suggestion
            raise ValueError(f"budget type must be `epochs` or `runtime`, but got {self.budget_type}")
```

**Review by nabenabe0928 — CHANGES_REQUESTED** [review]
Thanks for the change.
I added two minor suggestions.

File: autoPyTorch/evaluation/abstract_evaluator.py [hunk 3]
```diff
@@ -533,13 +519,65 @@ def __init__(self, backend: Backend,
...
+        self.fit_dictionary = {**pipeline_config, **self.fit_dictionary}
+        # If the budget is epochs, we want to limit that in the fit dictionary
+        if self.budget_type == 'epochs':
+            self.fit_dictionary['epochs'] = self.budget
+            self.fit_dictionary.pop('runtime', None)
+        elif self.budget_type == 'runtime':
+            self.fit_dictionary['runtime'] = self.budget
+            self.fit_dictionary.pop('epochs', None)
+        else:
+            raise ValueError(f"Unsupported budget type '{self.budget_type}' provided")
```
Comment (ravinkohli) [inline autoPyTorch/evaluation/abstract_evaluator.py:579]:
```suggestion
            raise ValueError(f"budget type must be `epochs` or `runtime`, but got {self.budget_type}")
```

**Pull request merged**